### PR TITLE
Fix bug in HyperParameterScheduler

### PR DESCRIPTION
**Pull request by jneeven**
It would previously crash when used when not specifying an optimizer, since `self.model` does not yet exist in ` __init__`. `set_model` is called once as part of `model.fit`, so this seems the best place for it.

### CODE REVIEW

**Comment by jneeven** [comment]
I accidentally removed the `LogHistory` from the `callbacks_test`, but it wasn't used anyway. I discussed with @lgeiger and @leonoverweel  and we concluded it should be fine to remove it for now.

File: larq/callbacks.py [hunk 1]
```diff
@@ -24,19 +27,32 @@ class HyperparameterScheduler(tf.keras.callbacks.Callback):
...
     """
 
-    def __init__(self, schedule, hyperparameter, optimizer=None, verbose=0):
+    def __init__(
+        self,
+        schedule: Callable,
+        hyperparameter: str,
+        optimizer: Optional[keras.optimizers.Optimizer] = None,
+        verbose: Optional[int] = 0,
+    ):
```
Comment (lgeiger) [inline larq/callbacks.py]: Thanks for the typehints. Unfortunately they don't show up nicely in our docs. I guess I need to refine our API docs generation. Though I think this shouldn't block this PR.

Comment (leonoverweel) [reply larq/callbacks.py]: @lgeiger looks like pydoc-markdown [doesn't have support for this yet](https://github.com/NiklasRosenstein/pydoc-markdown/issues/86).

Comment (lgeiger) [reply larq/callbacks.py]: We could also try out https://github.com/keras-team/keras-autodoc which looks similar.

Comment (lgeiger) [reply larq/callbacks.py]: For now we simply ignore the typehints in our rendered docs. I made some quick changes in to https://github.com/lgeiger/pydoc-markdown/commit/d6628ebd9f2ad5429c6898e63e5626873bdfc631

**Pull request merged**